Pocket/pocket-ios
# feat(widgets): add initial analytics

## Summary
Add analytics for widgets
* Track when widget is tapped and opens the application
* Track the current count and details of widget that a user has

## References 
IN-1464

## Implementation Details
Add two analytic events `widgetOpenEngagement` and `widgetDetailsEngagement`

## Test Steps
1. Add a widget and tap on it. Verify that the `widgetOpenEngagement` is triggered.
2. Add multiple widgets and launch the app. Verify that the `widgetDetailsEngagement` is triggered.

## PR Checklist:
- N/A Added Unit / UI tests

## Screenshots
